### DISCUSSION TITLE
using decontx with filtered or raw matrix?

### DISCUSSION THREAD

**Discussion by dm8000** [campbio/celda#375] [Q&A]
Hello

There are several QC steps in my pipeline, such as
1-cellranger filtering based on number of features;
2-tags with minimum and maximal features, percentage of mitochondrial genes, etc;
3-doublet removal

I see none of these steps are done in the vignette, so I assume I should run decontx before running any of these steps? even if there is a large number of apparently empty droplets in my matrix?

**Comment by joshua-d-campbell** [comment]
Hello @dm8000,
The main matrix to input into decontX is the cellranger filtered matrix, which filters out empty droplets from the raw matrix. DecontX can be run in two ways: with or without inputing the raw matrix into the `background` parameter. Generally, you don't need to use the background parameter unless any type of sorting was done and the RNA for some cell types are in the ambient material within the empty drops that are not found in the actual cells within filtered matrix.

You should run decontX before filtering for other QC metrics like min/max features, mitochondrial percentage, and doublets. Also, decontX can be run before any feature filtered. We also have a streamlined QC pipeline that runs a bunch of QC tools including decontX in the package [singleCellTK](https://camplab.net/sctk/v2.7.1/articles/cnsl_cellqc.html).

FYI, I'm going to move this to a Discussion rather than an Issue so others can see it in the future.

**Reply by vwtlin** [reply]
Hi @joshua-d-campbell, thanks for the amazing tool! I am wondering if you recommend to run DoubletFinder on the output generated by decontX. The singleCellTK package seems to run these two tools in parallel, please correct me if I misunderstand this.

Cheers,
Weitao

**Comment by joshua-d-campbell** [comment]
Hi @vwtlin, I would recommend to run DoubletFinder on the original uncorrected matrix (at least at first). Sometimes DecontX can flag doublets as having high contamination and may remove it thus making the cell not look like a doublet to other tools.

**Reply by vwtlin** [reply]
Hi @joshua-d-campbell, thank you so much for your prompt reply and advice! I didn't realise you were also answering the questions in singlecellTK. I still have some questions and I will pool them together under that thread. Thanks again!

Cheers,
Weitao
